docs/engine/Life-of-a-Flutter-Frame.md

```diff
@@ -41,6 +40,27 @@ Once a surface is acquired, the [LayerTree][layerTree] is rasterized to the surf
 
 The above process is repeated until the pipeline is empty.
 
+## Warm-up frame
+
+Normally, the Flutter framework begins producing a frame when it receives
+a vsync event from the operating system. However, this may not happen for
+several milliseconds after the app starts (or after a hot reload). To make
+use of the time between when the widget tree is first configured and when
+the engine requests an update, the framework schedules a _warm-up frame_
+using [PlatformDispatcher.scheduleWarmUpFrame][scheduleWarmUpFrame].
+
+A warm-up frame may never actually render (as the engine did not request
+it and therefore does not have a valid context in which to paint), but it
+will cause the framework to go through the steps of building, laying out,
+and painting, which can together take several milliseconds. Thus, when the
+engine requests a real frame, much of the work will already have been
+completed, and the framework can generate the frame with minimal
+additional effort.
+
+At startup, a warm-up frame can be produced before the Flutter engine has reported the
+initial view metrics using [PlatformDispatcher.onMetricsChanged][onMetricsChanged].
+As a result, the first frame can be produced with a size of zero.
```

Comment:
Is this paragraph really needed? Is it something that really confuses people?

Reply:
We can probably just count this case as another case of "the warm up frame is not presented at all".

Reply:
Hey there, I agree this paragraph is a bit deep in the weeds. But this topic is the reason why I sent this pull request, as a follow-up to answering this question: https://github.com/flutter/flutter/issues/149974

Reply:
I see, thanks for explanation!